sdk/keyvault/keyvault-common/src/challengeBasedAuthenticationPolicy.ts

```diff
@@ -156,25 +171,103 @@ export function createKeyVaultChallengeCallbacks(
 
     const accessToken = await getAccessToken([scope], {
       ...getTokenOptions,
+      enableCae: true,
       tenantId: parsedChallenge.tenantId,
     });
 
     if (!accessToken) {
-      return false;
+      // No access token provided, treat as no-op
+      return response;
     }
 
     request.headers.set("Authorization", `Bearer ${accessToken.token}`);
 
     challengeState = {
       status: "complete",
       scopes: [scope],
+      tenantId: parsedChallenge.tenantId,
+    };
+
+    // We have a token now, so try send the request again
+    return next(request);
+  }
+
+  async function handleCaeChallenge(
+    request: PipelineRequest,
+    response: PipelineResponse,
+    next: SendRequest,
+  ): Promise<PipelineResponse> {
+    // Cannot handle CAE challenge if a regular challenge has not been completed first
+    if (challengeState.status !== "complete") {
+      return response;
```

Comment:
Should we throw instead? It should be impossible to land here because presenting an access token is a prerequisite for receiving a CAE challenge. The client shouldn't be able to acquire a token without first discovering auth parameters by handling an ordinary challenge. Landing here would then be a bug, no?

Reply:
In my opinion, no. At this point in the function we are still establishing whether the response from the service is a CAE challenge (if it is a challenge at all). If it's not a valid CAE challenge, like in the case you describe, then we don't try and handle it and the 401 is left unhandled. Ultimately this will result in Core throwing an error anyway.

In theory it would be possible to land here if, we received a 200 (or really anything that's not a 401) from the service before receiving a regular non-CAE challenge. I'm not sure if that's possible, probably not, but it does seem more correct to me to not throw in that case. That said, if you feel strongly about this, I can change it.